nomad/stream/event_broker.go

```diff
@@ -367,6 +367,12 @@ func aclAllowsSubscription(aclObj *acl.ACL, subReq *SubscribeRequest) bool {
 			if ok := aclObj.AllowNodeRead(); !ok {
 				return false
 			}
+		case structs.TopicNodePool:
+			// Require management token for node pools since we can't filter
+			// out node pools the token doesn't have access to.
+			if ok := aclObj.IsManagement(); !ok {
+				return false
+			}
```

Comment:
A management token here feels like an overkill, but I couldn't find a more granular way to check this. One option could be to check for `nomad_poll "*"` read capability or create a new capability that can only be used with `*`, but these felt a bit off.